source/Hazard.cpp

```diff
@@ -0,0 +1,120 @@
+/* Hazard.cpp
+Copyright (c) 2020 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Hazard.h"
+
+#include "DataNode.h"
+#include "Effect.h"
+#include "GameData.h"
+#include "Random.h"
+
+using namespace std;
+
+
+void Hazard::Load(const DataNode &node)
+{
+	if(node.Size() < 2)
+		return;
```

Comment:
Just as an aside, I've noticed that there's a lack of consistency among a lot of Load functions across the various classes. Some have an error if there is no name, others just return. Some use a "key" that is `child.Token(0)` then check against that, or use a `hasValue` boolean and a `value` double, others don't. Might be worth organizing all that at some point.